source/common/config/filter_json.cc

```diff
@@ -336,5 +327,56 @@ void FilterJson::translateTcpProxy(const Json::Object& json_tcp_proxy,
   }
 }
 
+void FilterJson::translateTcpRateLimitFilter(
+    const Json::Object& json_config, envoy::api::v2::filter::network::RateLimit& proto_config) {
+  json_config.validateSchema(Json::Schema::RATELIMIT_NETWORK_FILTER_SCHEMA);
+
+  JSON_UTIL_SET_STRING(json_config, proto_config, stat_prefix);
+  JSON_UTIL_SET_STRING(json_config, proto_config, domain);
+  JSON_UTIL_SET_DURATION(json_config, proto_config, timeout);
+
+  auto* descriptors = proto_config.mutable_descriptors();
+  for (const auto& json_descriptor : json_config.getObjectArray("descriptors", false)) {
+    auto* descriptor = descriptors->Add();
+    auto* entries = descriptor->mutable_entries();
+    for (const auto& json_entry : json_descriptor->asObjectArray()) {
+      auto* entry = entries->Add();
+      JSON_UTIL_SET_STRING(*json_entry, *entry, key);
+      JSON_UTIL_SET_STRING(*json_entry, *entry, value);
+    }
+  }
+}
+
+void FilterJson::translateHttpRateLimitFilter(
+    const Json::Object& json_config, envoy::api::v2::filter::http::RateLimit& proto_config) {
+  json_config.validateSchema(Json::Schema::RATE_LIMIT_HTTP_FILTER_SCHEMA);
+
+  JSON_UTIL_SET_STRING(json_config, proto_config, domain);
+  proto_config.set_stage(json_config.getInteger("stage", 0));
+
+  JSON_UTIL_SET_STRING(json_config, proto_config, request_type);
+  JSON_UTIL_SET_DURATION(json_config, proto_config, timeout);
+}
+
+void FilterJson::translateClientSslAuthFilter(
+    const Json::Object& json_config, envoy::api::v2::filter::network::ClientSSLAuth& proto_config) {
+  json_config.validateSchema(Json::Schema::CLIENT_SSL_NETWORK_FILTER_SCHEMA);
+
+  JSON_UTIL_SET_STRING(json_config, proto_config, auth_api_cluster);
+  JSON_UTIL_SET_STRING(json_config, proto_config, stat_prefix);
+  JSON_UTIL_SET_DURATION(json_config, proto_config, refresh_delay);
+
+  auto* ip_white_list = proto_config.mutable_ip_white_list();
+  for (const std::string& cidr_str : json_config.getStringArray("ip_white_list", true)) {
+    auto* entry = ip_white_list->Add();
+    auto split_index = cidr_str.find("/");
+    entry->set_address_prefix(cidr_str.substr(0, split_index));
+    if (split_index != std::string::npos) {
+      const char* prefix_len_str = cidr_str.substr(split_index + 1, std::string::npos).c_str();
+      entry->mutable_prefix_len()->set_value(std::atoi(prefix_len_str));
```

Comment:
There is already CIDR stuff here: https://github.com/envoyproxy/envoy/blob/master/source/common/config/address_json.h#L32